pkg/util/ctx_test.go

```diff
@@ -56,8 +60,132 @@ func (s *ctxValueSuite) TestCanceledContext(c *check.C) {
 func (s *ctxValueSuite) TestChangefeedIDNotSet(c *check.C) {
 	defer testleak.AfterTest(c)()
 	c.Assert(ChangefeedIDFromCtx(context.Background()), check.Equals, "")
+	changefeedID := ChangefeedIDFromCtx(context.Background())
+	c.Assert(changefeedID, check.Equals, "")
 	ctx := context.WithValue(context.Background(), ctxKeyChangefeedID, 1321)
-	c.Assert(ChangefeedIDFromCtx(ctx), check.Equals, "")
+	changefeedID = ChangefeedIDFromCtx(ctx)
+	c.Assert(changefeedID, check.Equals, "")
+}
+
+func (s *ctxValueSuite) TestShouldReturnTimezone(c *check.C) {
+	defer testleak.AfterTest(c)()
+	tz, _ := getTimezoneFromZonefile("UTC")
+	ctx := PutTimezoneInCtx(context.Background(), tz)
+	tz = TimezoneFromCtx(ctx)
+	c.Assert(tz.String(), check.Equals, "UTC")
+}
+
+func (s *ctxValueSuite) TestTimezoneNotSet(c *check.C) {
+	defer testleak.AfterTest(c)()
+	tz := TimezoneFromCtx(context.Background())
+	c.Assert(tz, check.IsNil)
+	ctx := context.WithValue(context.Background(), ctxKeyTimezone, 1321)
+	tz = TimezoneFromCtx(ctx)
+	c.Assert(tz, check.IsNil)
+}
+
+func (s *ctxValueSuite) TestShouldReturnTableInfo(c *check.C) {
+	defer testleak.AfterTest(c)()
+	ctx := PutTableInfoInCtx(context.Background(), 1321, "ello")
+	tableID, tableName := TableIDFromCtx(ctx)
+	c.Assert(tableID, check.Equals, int64(1321))
+	c.Assert(tableName, check.Equals, "ello")
+}
+
+func (s *ctxValueSuite) TestTableInfoNotSet(c *check.C) {
+	defer testleak.AfterTest(c)()
+	tableID, tableName := TableIDFromCtx(context.Background())
+	c.Assert(tableID, check.Equals, int64(0))
+	c.Assert(tableName, check.Equals, "")
+	ctx := context.WithValue(context.Background(), ctxKeyTableID, 1321)
+	tableID, tableName = TableIDFromCtx(ctx)
+	c.Assert(tableID, check.Equals, int64(0))
+	c.Assert(tableName, check.Equals, "")
+}
+
+func (s *ctxValueSuite) TestShouldReturnKVStorage(c *check.C) {
+	defer testleak.AfterTest(c)()
+	kvStorage := newMockStorage()
```

Comment:
We can reuse TiDB lib to generate a `kv.Storage` without rewriting a new one, such as

https://github.com/pingcap/tidb/blob/8071a90d3621553559885954b3e7a8ef48788c6a/store/store_test.go#L60

Reply:
I did not find this lib before. Thank you!